src/columnar_transposition.rs

```diff
@@ -10,56 +10,88 @@ use common::{alphabet, keygen};
 use common::alphabet::Alphabet;
 
 /// A Columnar Transposition cipher.
-///
 /// This struct is created by the `new()` method. See its documentation for more.
 pub struct ColumnarTransposition {
     key: String,
+    null_char: char,
+    use_nulls: bool,
 }
 
 impl Cipher for ColumnarTransposition {
-    type Key = String;
+    type Key = (String, String);
     type Algorithm = ColumnarTransposition;
 
-    /// Initialize a Columnar Transposition cipher given a specific key.
+    /// Initialize a Columnar Transposition cipher given:
+    /// * a specific `key`, and
+    /// * a specified `null` - a character that will pad the columns
     ///
     /// Will return `Err` if one of the following conditions is detected:
     ///
     /// * The `key` length is 0.
     /// * The `key` contains non-alphanumeric symbols.
     /// * The `key` contains duplicate characters.
-    fn new(key: String) -> Result<ColumnarTransposition, &'static str> {
-        keygen::columnar_key(&key)?;
-        Ok(ColumnarTransposition { key: key })
+    /// * The `null` contains more than one character
+    /// * The `null` contains a character in the `key`
+    fn new(key: (String, String)) -> Result<ColumnarTransposition, &'static str> {
+        keygen::columnar_key(&key.0)?;
+
+        let mut use_nulls = key.1.len() == 1;
+        let mut null_char = '\u{0}'; // Default null
+
+        if use_nulls {
```

Comment:
I suggest you read up on the `Option<>` type. A lot of this processing code could be simplified.

Reply:
This sounds a plan. I thought the code was messy here and there must be another, better way of doing it. 

So, for now, reject the PR and I'll resubmit when I've done some reading ;)